### ISSUE TITLE
Validation for incorrect LocalDate

### ISSUE THREAD

**Issue by OliveraM** [donbeave/graphql-java-datetime#34]
I have the spring project with dependencies:
- GraphQL Java Kickstart
- and latest version of graphql-datetime-spring-boot-starter
- defined date format is 'yyyy-MM-dd'
But when I pass incorrect date '2020-53-23', there is no error.

I would expect to get an error in the response instead of empty object. Is there the way to accomplish this?

**Comment by donbeave** [comment]
@OliveraM yes, you are right. Actually, I thought it will return the validation error, but it's not. I didn't touch this library for a while, if you have time to dig into the root which causes this issue and fix it, I will be happy to merge PR with such changes.

**Comment by setchy** [comment]
I just tried to recreate this scenario and for me it correctly threw an error.  I also tried added new tests to GraphQLLocalDateTest and it also seemed to handle this scenario

```
{
  "errors": [
    {
      "message": "Can't serialize value (/testDate) : Invalid value '2020-55-01' for LocalDate",
      "path": [
        "testDate"
      ],
      "extensions": {
        "classification": "DataFetchingException"
      }
    }
  ],
  "extensions": {
    "ftv1": "GgsI7YCNgAYQ4MvzLyILCO2AjYAGEKjA6gdY+ImJKHIoYiYKCHRlc3REYXRlGglMb2NhbERhdGVA4trkFUjq5qceagVRdWVyeQ=="
  },
  "data": {
    "testDate": null
  }
}
```

**Comment by donbeave** [comment]
@setchy you can reproduce it with `graphql-datetime-sample-app`. There is no validation for input fields.

My request is:

```graphql
mutation {
  plusDay(input: "2020-22-22")
}
```

And it will successfully pass to the resolver but with a `null` value instead:

<img width="1065" alt="image" src="https://user-images.githubusercontent.com/139017/104923292-d10b8a80-59d6-11eb-9c8e-e6ddc9c05a7b.png">

In this particular case, it will throw an NPE inside the resolver, that's why the result will have an error:

<img width="1494" alt="image" src="https://user-images.githubusercontent.com/139017/104923466-0dd78180-59d7-11eb-8a93-c9195ae3c298.png">

But in the case where the input value is not required, it will look the same as not value was provided, which is incorrect I think.

**Comment by setchy** [comment]
You're right @donbeave  - I understood the initial question as trying to passing an invalid date from the model through the resolver, not via a mutation or query input argument.

**Comment by setchy** [comment]
I think I've got a fix for this.  Working on updating the unit-tests now

**Comment by setchy** [comment]
WIP, but here is the initial commit.  Appreciate some early feedback

https://github.com/setchy/graphql-java-datetime/tree/feature/issue-34-errorhandling

Like I said, need to update quite a few unit-tests as parseLiteral is now throwing the CoercsingParseLiteralException

**Comment by donbeave** [comment]
I like it, looks good to me.

**Comment by setchy** [comment]
@OliveraM see #37

**Comment by donbeave** [comment]
Fix released in 4.0.0 version: https://bintray.com/donbeave/maven/graphql-java-datetime/4.0.0
